src/main/java/redis/clients/jedis/util/JedisURIHelper.java

```diff
@@ -70,6 +71,73 @@ public static RedisProtocol getRedisProtocol(URI uri) {
     return null; // null (default) when not defined
   }
 
+  private static final Integer ZERO_INTEGER = 0;
+
+  public static ClientSideCache getClientSideCache(URI uri) {
+    if (uri.getQuery() == null) return null;
+
+    boolean guava = false, caffeine = false; // cache_lib
+    Integer maxSize = null; // cache_max_size --> 0 = disbale
+    Integer ttl = null; // cache_ttl --> 0 = no ttl
+    // cache-max-idle
+
+    String[] params = uri.getQuery().split("&");
+    for (String param : params) {
+      int idx = param.indexOf("=");
+      String key = param.substring(0, idx);
+      String val = param.substring(idx + 1);
+
+      switch (key) {
+
+        case "cache_lib":
+          switch (val) {
+            case "guava":
+              guava = true;
+              break;
+            case "caffeine":
+              caffeine = true;
+              break;
+            default:
+              throw new IllegalArgumentException("Unsupported library " + val);
+          }
+          break;
+
+        case "cache_max_size":
+          maxSize = Integer.parseInt(val);
```

Comment:
Same here at parsing. There are typos and stuff, better catch `NumberFormatException` and return a nice error?